### DISCUSSION TITLE
Gosund WB4

### DISCUSSION THREAD

**Discussion by davidebeatrici** [arendst/Tasmota#15879] [Q&A] [answered]
Bought on [Amazon.it](https://www.amazon.it/gp/product/B086M7784V) in September 2020, unfortunately the link doesn't work anymore.
The official page should be https://us.gosund.com/products/app-smart-bulb-wb4.

---

The main board is identified as `WB2_Main(20181022)_V0.5`.

<details>
<summary>ESP board</summary>

![](https://user-images.githubusercontent.com/5897523/175833556-298d4ab7-1bac-4fa9-8710-b355967b82a6.png)
</details>

[ESP8266EX](https://espressif.com/sites/default/files/documentation/0a-esp8266ex_datasheet_en.pdf).

According to my multimeter in continuity mode, TX and RX are the 2nd and 3rd pin from the left.

Unfortunately, the serial console only shows noise and I made sure to use the same ground as the power supply.
Interestingly, the output appears to be the same even after multiple reboots.

---

Possibly related:

- #10389
- blakadder/templates#653
- https://www.reddit.com/r/tasmota/comments/iu6a13/help_identifying_pins_on_this_light_bulb
  The board appears to be identical, even if the Imgur album is called `Gosund WB3`.
  <details>
  <summary>Photo</summary>

  ![](https://i.imgur.com/cNBub4Q.jpeg)
  </details>

**Comment by sfromis** [comment, marked as answer]
Serial "noise" may be up to what baud rate you selected, and possibly what mode it booted in. You should usually use 115200, or maybe 74880.

**Reply by davidebeatrici** [reply]
Shouldn't a lower baud rate be more reliable and less susceptible to disturbances?

**Reply by sfromis** [reply]
Only if you can have the baud rate selection done on both ends at the same time. For boot messages, that means that the answer is a firm **no**.

While some terminal programs might handle wrong baud rate better than others, serial data being sampled with a different bit rate means garbled data.

**Reply by davidebeatrici** [reply]
You were absolutely right, `74880` is the correct baud rate in this case.

**Comment by davidebeatrici** [comment]
The Gosund WB3 template works!

https://templates.blakadder.com/gosund_WB3.html

**Reply by davidebeatrici** [reply]
Perhaps "WB4" is only the name of the ESP8266 board and what I have is a Gosund WB3?

@antoniomolram
